src/main/kotlin/com/sourcegraph/cody/inspections/CodyFixHighlightPass.kt

```diff
@@ -106,15 +116,18 @@ class CodyFixHighlightPass(val file: PsiFile, val editor: Editor) :
           if (progress.isCanceled) {
             break
           }
+          if (highlight.startOffset > document.textLength ||
+              highlight.endOffset > document.textLength ||
+              highlight.startOffset > highlight.endOffset) {
+            break
+          }
+
           val range = document.codyRange(highlight.startOffset, highlight.endOffset)
 
           if (myRangeActions.containsKey(range)) {
             continue
           }
-          val location =
-              ProtocolLocation(
-                  uri = protocolTextDocument.uri,
-                  range = document.codyRange(highlight.startOffset, highlight.endOffset))
+          val location = ProtocolLocation(uri = protocolTextDocument.uri, range = range)
```

Comment:
Perhaps we can get rid of that whole `protocolTextDocument`?
It seems we need it only for uri, and we can get is using `ProtocolTextDocument::uriFor`.
It is not marked as requiring EDT so there is a good chance we will also not need that future, etc..

Reply:
btw, doCollectInformation is called from the background thread by IntelliJ

![image](https://github.com/user-attachments/assets/de40a771-1add-47f2-b339-9c943b9b1b83)
(we are extending HighlightPass and collectInformation calls doCollectInformation)